horovod/keras/__init__.py

```diff
@@ -51,19 +52,7 @@ def get_gradients(self, loss, params):
         allreduce the gradients before returning them.
         """
         gradients = super(self.__class__, self).get_gradients(loss, params)
-        if hvd.size() > 1:
-            averaged_gradients = []
-            with tf.name_scope(self._name + "_Allreduce"):
-                for grad in gradients:
-                    if grad is not None:
-                        avg_grad = hvd.allreduce(grad, device_dense=self._device_dense,
-                                                 device_sparse=self._device_sparse)
-                        averaged_gradients.append(avg_grad)
-                    else:
-                        averaged_gradients.append(None)
-                return averaged_gradients
-        else:
-            return gradients
+        return _impl.get_gradients(gradients, self._device_dense, self._device_sparse, self._name)
```

Comment:
I think we can move this whole `_DistributedOptimizer` implementation class as well as chunk of DistributedOptimizer in impl, and construct it within a function that implements DistributedOptimizer.

Reply:
Done.  Moved to `impl.py`, but note that it changes the module of the derived optimizer (see `test_keras.py` changes).